src/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableArray`1.cs

```diff
@@ -602,11 +610,6 @@ public ImmutableArray<T> InsertRange(int index, ImmutableArray<T> items)
         public ImmutableArray<T> Add(T item)
         {
             var self = this;
-            if (self.Length == 0)
-            {
-                return ImmutableArray.Create(item);
-            }
-
```

Comment:
It's true that Insert does a length check, but that doesn't necessarily mean this should be eliminated.  For example, if Add can typically be inlined, Insert can't, and the case of adding to an empty list is common (which I believe it is), then this could actually be hurting performance in a common case because it forces the calling of a non-inlined method.  We shouldn't be making/taking changes like this without a solid set of perf measurements across a range of scenarios to demonstrate that it's actually a win.